src/Symfony/Component/Serializer/Normalizer/NormalizerDumperTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Exception\CircularReferenceException;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+
+trait NormalizerDumperTrait
+{
+    private function checkCircularReference($object, &$context)
+     {
+        $objectHash = spl_object_hash($object);
+        if (isset($context[ObjectNormalizer::CIRCULAR_REFERENCE_LIMIT][$objectHash])) {
+            throw new CircularReferenceException('A circular reference has been detected (configured limit: 1).');
+        } else {
+            $context[ObjectNormalizer::CIRCULAR_REFERENCE_LIMIT][$objectHash] = 1;
+        }
+    }
+
+    private function isAllowedAttribute($property, $attributeGroups, &$context) {
```

Comment:
Really no idea, I just like references but there's not point here I'll remove it.